finetuner/tuner/keras/miner.py

```diff
@@ -0,0 +1,153 @@
+from itertools import combinations, groupby, permutations
+from typing import Tuple
+
+import tensorflow as tf
+
+from ..base import BaseClassMiner, BaseSessionMiner
+
+
+class SiameseMiner(BaseClassMiner[tf.Tensor]):
```

Comment:
I still think that the Miner design is weird, I do not understand why it has to be tight to a `framework`. 

It should be depeding on labels, any embeddings and distances callables

Reply:
Computing triplets is significantly faster for large batches when using GPU - which requires the use of tensors.

Reply:
To expand on this - for class dataset this is a problem with n^3 time complexity, which is highly parallelizable. If `n=512` I got very significant time savings. Keeping everything on CPU/numpy would hinder the performance when training on GPU

Reply:
The usage of tensors and the abstractions I do not see why they are related.

Reply:
My concern is the explosion of work to add a new mining strategy, for instance some hard mining logic, I will have to implement it for each framework. Beyond, it surprises me that computing the triplets becoming the bottleneck of the system.